worlds/oot_soh/location_access/dungeons/ganons_castle.py

```diff
@@ -5,22 +5,350 @@
+        (Locations.GANONS_CASTLE_TOWER_BOSS_KEY_CHEST, lambda bundle: can_kill_enemy(bundle, Enemies.STALFOS, EnemyDistance.CLOSE, True, 2))
+    ])
+    # Connections
+    connect_regions(Regions.GANONS_TOWER_FLOOR_2, world, [
+        (Regions.GANONS_TOWER_FLOOR_1, lambda bundle: can_kill_enemy(bundle, Enemies.STALFOS, EnemyDistance.CLOSE, True, 2)),
+        (Regions.GANONS_TOWER_FLOOR_3, lambda bundle: can_kill_enemy(bundle, Enemies.STALFOS, EnemyDistance.CLOSE, True, 2))
+    ])
+    
+    ## Ganon's Tower Floor 3
+    # Connections
+    connect_regions(Regions.GANONS_TOWER_FLOOR_3, world, [
+        (Regions.GANONS_TOWER_FLOOR_2, lambda bundle: can_kill_enemy(bundle, Enemies.IRON_KNUCKLE, EnemyDistance.CLOSE, True, 2)),
+        (Regions.GANONS_TOWER_BEFORE_GANONDORFS_LAIR, lambda bundle: can_kill_enemy(bundle, Enemies.IRON_KNUCKLE, EnemyDistance.CLOSE, True, 2))
+    ])
+    
+    ## Ganon's Tower Before Ganondorf's Lair
+    # Locations
+    add_locations(Regions.GANONS_TOWER_BEFORE_GANONDORFS_LAIR, world, [
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT1, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT2, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT3, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT4, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT5, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT6, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT7, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT8, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT9, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT10, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT11, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT12, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT13, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT14, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT15, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT16, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT17, lambda bundle: can_break_pots(bundle)),
+        (Locations.GANONS_CASTLE_GANONS_TOWER_POT18, lambda bundle: can_break_pots(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.GANONS_TOWER_BEFORE_GANONDORFS_LAIR, world, [
+        (Regions.GANONS_TOWER_FLOOR_3, lambda bundle: True),
+        (Regions.GANONDORFS_LAIR, lambda bundle: has_item(Items.GANONS_CASTLE_BOSS_KEY, bundle))
+    ])
```

Comment:
You couldn't have known this, but this also needs an `or` condition for triforce hunt being on. this may be a little unintuitive but AP wants all locations to be reachable, but with triforce hunt on the boss key simply doesn't exist in the item pool. Instead, we also have another check on killing ganon for triforce hunt being `off` or already having gotten the game completion event.

tl:dr; needs an `or` condition for triforce hunt being on